src/keycloak/chart/templates/statefulset.yaml

```diff
@@ -113,15 +113,17 @@ spec:
             # Dumb value (not used in the nginx provider, but required by the SPI)
             - name: KC_SPI_X509CERT_LOOKUP_NGINX_SSL_CLIENT_CERT_CHAIN_PREFIX
               value: UNUSED
-          {{- if .Values.devMode }}
-            # Enable dubug logs in dev mode
+          {{- if or .Values.devMode .Values.debug }}
+            # Enable debug logs
             - name: KC_LOG_LEVEL
               value: DEBUG
             - name: QUARKUS_LOG_CATEGORY__ORG_APACHE_HTTP__LEVEL
               value: DEBUG
             - name: QUARKUS_LOG_CATEGORY__ORG_KEYCLOAK_SERVICES_X509__LEVEL
               value: TRACE
-          {{- else }}
+          {{- end }}    
+          {{- if .Values.postgresql.host }}
+          {{- if .Values.devMode -}}{{fail "Cannot use devMode when an external PostgreSQL Database (postgres.host) is configured."}}{{- end -}}
```

Comment:
Would it make sense to add a helper that checks if host, user, pass, database, and port are all set? This would generally work, but maybe is better to strictly enforce all 5 values being passed in for this config?

Another scenario I just thought of...if someone disabled devMode but did not configure postgres that may also be worth failing on?

Reply:
Agreed - a helper function that checks for all of the cases mentioned would be the best solution.

Reply:
@mjnagel just pushed up some changes that moves all of the logic into a helper function (`keycloak.postgresql.config`). The function makes all config for `postgresql` conditional if all of the requirements mentioned are satisfied. LMK if you want me to walk you through the test cases I ran.